DOC-1759: Updated end_container_on_empty_block (#2468)
tinymce/tinymce-docs: modules/ROOT/partials/configuration/end_container_on_empty_block.adoc

[[end_container_on_empty_block]]
== `+end_container_on_empty_block+`

> This option allows you to split the current container block element and specify the type of element created if the enter key is pressed inside an empty inner block element.

Reviewer: This change is incorrect... it doesn't let you specify the type of element created. When using a string it lets you specify which blocks should be split when the enter key is pressed.

Reviewer: We also need to include somewhere that the string value only works in 6.1 or above.

> This option allows you to split the current container block element if the enter key is pressed inside an empty inner block element, and specify the type of element created.

Reviewer 2: ```suggestion
This option allows you to split the current container block element and specify the type of element created if the enter key is pressed inside an empty inner block element.
```

I moved the two option-related phrases together.

The original has an unstated but easily inferred ‘then’ adverb. That is, it’s very easy to read the original as meaning

> This option allows you to split the current container block
> element if the enter key is pressed inside an empty inner block
> element, and _then_ specify the type of element created.

This reading — which makes the inferred causal relationship explicit — turns the option into something that acts as follows:

To use `end_container_on_empty_block` to split the current container

1. put the insertion point inside an empty inner block element

then

2. press *Enter*

then

3. specify the type of element to be created.

Given the option is set with a pre-chosen value, this inferred causal relationship is incorrect.

And moving the phrase makes this inference much harder to make.

*Type:* `+Boolean+` or `+String+`

*Default value:* `+'blockquote'+`

> *Possible values:* `+true+`, `+false+`, `+String+`

Reviewer: This isn't correct... it's not any string, it's a comma separated list of element names.
```suggestion
*Possible values:* `+true+`, `+false+` or a comma separated `+String+` of element tag names
```

[[end_container_on_empty_block]]
== `+end_container_on_empty_block+`

> This option allows you to split the current container block element when the enter key is pressed, and specify which blocks should be split.

Reviewer: This change still isn't correct sorry @MitchCTiny. It's not something like allows in addition to, it's one or the other.
```suggestion
This option allows specifying how the current container block element is split when the enter key is pressed. It can either, always split the block, split for specific block elements or not split at all.
```

I think this should cover it better, but not too sure on the wording so I'd like to get @brian-forte's input here.

> This option allows you to split the current container block element when the enter key is pressed, and specify which blocks should be split.

Reviewer 2: A minor edit to @lnewson’s edit plus some presentational changes:
```suggestion
This option specifies how the current container block element is split when the Enter key is pressed inside an empty inner block element. The available options are:

* always split the block;
* split for specific block elements; or
* do not split the block at all.
```
When discussing a lists of options, it’s always worth considering if presenting said options as a literal list helps.

I think it does here. The first sentence above document how and when. And the second sentence sets out what, and lays out the what in a list for clarity’s sake.

[[end_container_on_empty_block]]
== `+end_container_on_empty_block+`

This option specifies how the current container block element is split when the Enter key is pressed inside an empty inner block element. The available options are:

* always split the block;
* split for specific block elements; or
* do not split the block at all.

*Type:* `+Boolean+` or `+String+`

*Default value:* `+'blockquote'+`

> *Possible values:* `+true+`, `+false+` or comma separated `+String+` of element tag names

Reviewer: Oops sorry Mitch, just noticed I had a grammar error in the suggestion I gave previously 😞 
```suggestion
*Possible values:* `+true+`, `+false+` or a comma separated `+String+` of element tag names
```

NOTE: Prior to {productname} 6.1, this option only accepted `+Boolean+` values.

=== Example: Using `+end_container_on_empty_block+`

[source,js]
----
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML

>   end_container_on_empty_block: 'blockquote'

Reviewer: This change shouldn't have been done as we're supposed to show a different value to the default in the examples.
```suggestion
  end_container_on_empty_block: true
```
